modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java

```diff
@@ -101,7 +103,7 @@ Function<String, ZonedDateTime> getFunction(String format, ZoneId zoneId, Locale
                     ZonedDateTime newTime = Instant.EPOCH.atZone(ZoneOffset.UTC).withYear(year);
                     for (ChronoField field : FIELDS) {
                         if (accessor.isSupported(field)) {
-                            newTime = newTime.with(field, accessor.get(field));
+                            newTime = newTime.with(field, accessor.getLong(field));
```

Comment:
The cautionary notes on `getLong` (`The returned value may be outside the valid range of values for the field.`) has me a bit spooked. I realize you changed it here because `.get` throws for `INSTANT_SECONDS` (which needs to return a long).

Do you object to special casing the `INSTANT_SECONDS` field in an if/else and having it call `getLong` because it must? Alternatively, I suppose you could convince me somehow that the warning on `getLong` can't possibly apply.